src/Computation/Values/Constants/ConstantUnsigned.cs

```diff
@@ -55,7 +55,12 @@ public ConstantDouble AsDouble()
 
     public override bool Equals(IValue? other)
     {
-        return other is IConstantValue v && Equal(v.AsUnsigned());
+        return Equals(other as IConstantValue);
+    }
+
+    public bool Equals(IConstantValue? other)
+    {
+        return other is not null && Equal(other.AsUnsigned());
     }
 
```

Comment:
I assume this gets `GetHashCode` automatically as a `record`.

Reply:
Yeah and in this case we do want the default. It's all the other Constants that call through to this one which is why I changed their behaviours and left this one. Same for Equals.